core/sdk-core/src/main/java/software/amazon/awssdk/core/client/builder/SdkDefaultClientBuilder.java

```diff
@@ -399,31 +442,24 @@ public final void setAsyncConfiguration(ClientAsyncConfiguration asyncConfigurat
 
     @Override
     public final B overrideConfiguration(ClientOverrideConfiguration overrideConfig) {
-        clientConfiguration.option(EXECUTION_INTERCEPTORS, overrideConfig.executionInterceptors());
-        clientConfiguration.option(RETRY_POLICY, overrideConfig.retryPolicy().orElse(null));
-        clientConfiguration.option(ADDITIONAL_HTTP_HEADERS, overrideConfig.headers());
-        clientConfiguration.option(SIGNER, overrideConfig.advancedOption(SIGNER).orElse(null));
-        clientConfiguration.option(USER_AGENT_SUFFIX, overrideConfig.advancedOption(USER_AGENT_SUFFIX).orElse(null));
-        clientConfiguration.option(USER_AGENT_PREFIX, overrideConfig.advancedOption(USER_AGENT_PREFIX).orElse(null));
-        clientConfiguration.option(API_CALL_TIMEOUT, overrideConfig.apiCallTimeout().orElse(null));
-        clientConfiguration.option(API_CALL_ATTEMPT_TIMEOUT, overrideConfig.apiCallAttemptTimeout().orElse(null));
-        clientConfiguration.option(DISABLE_HOST_PREFIX_INJECTION,
-                                   overrideConfig.advancedOption(DISABLE_HOST_PREFIX_INJECTION).orElse(null));
-        clientConfiguration.option(PROFILE_FILE, overrideConfig.defaultProfileFile().orElse(null));
-        clientConfiguration.option(PROFILE_NAME, overrideConfig.defaultProfileName().orElse(null));
-        clientConfiguration.option(METRIC_PUBLISHERS, overrideConfig.metricPublishers());
-        clientConfiguration.option(EXECUTION_ATTRIBUTES, overrideConfig.executionAttributes());
-        overrideConfig.advancedOption(ENDPOINT_OVERRIDDEN_OVERRIDE).ifPresent(value -> {
-            clientConfiguration.option(ENDPOINT_OVERRIDDEN, value);
-        });
-        overrideConfig.advancedOption(SIGNER).ifPresent(s -> clientConfiguration.option(SIGNER_OVERRIDDEN, true));
+        clientOverrideConfiguration = overrideConfig;
+
         return thisBuilder();
     }
 
     public final void setOverrideConfiguration(ClientOverrideConfiguration overrideConfiguration) {
         overrideConfiguration(overrideConfiguration);
     }
 
+    @Override
+    public final ClientOverrideConfiguration overrideConfiguration() {
+        if (clientOverrideConfiguration == null) {
+            return ClientOverrideConfiguration.builder().build();
+        }
+
+        return clientOverrideConfiguration.toBuilder().build();
```

Comment:
Do we need to do toBuilder().build()?

Reply:
Ah, I guess not, since its not possible to change the values with the configuration. I'll fix it.